v1.2.10: looks like some fields are embedded differently / incorrectly

@j2gg0s just upgraded to v1.2.10 and something is broken or changed in backwards incompatible way. I am tempted to blame https://github.com/uptrace/bun/commit/b410e420ab888d87d2b6ebb014f13baae8fdc2b7

The bug manifests as if the field type is detected incorrectly:

```
panic: reflect: call of reflect.Value.SetUint on string Value
goroutine 257 [running]:
panic({0x1bccd20?, 0x4000ccacc0?})
        runtime/panic.go:787 +0xf8
reflect.Value.SetUint({0x1b01560?, 0x4000ac5018?, 0x400174a4e8?}, 0x1)
        reflect/value.go:2184 +0x100
github.com/uptrace/bun/schema.scanUint64({0x1b01560?, 0x4000ac5018?, 0x1f347e0?}, {0x1b01720?, 0x2cb9fc8?})
        github.com/uptrace/bun@v1.2.10/schema/scan.go:196 +0x9c
github.com/uptrace/bun/schema.(*Field).ScanWithCheck(0x1d397c0?, {0x1b01560?, 0x4000ac5018?, 0x4000f4a408?}, {0x1b01720?, 0x2cb9fc8?})
        github.com/uptrace/bun@v1.2.10/schema/field.go:132 +0x90
```

For now I've rolled back to https://github.com/uptrace/bun/pull/1127 and all is back to normal.

--- a/schema/table.go
+++ b/schema/table.go
@@ -171,7 +171,7 @@ func (t *Table) processFields(typ reflect.Type) {
 					if _, ok := ebdStructs[k]; !ok {
 						ebdStructs[k] = &structField{
 							Index: makeIndex(sf.Index, v.Index),
-							Table: subtable,
+							Table: v.Table,
 						}
 					}
 				}

--- a/schema/table_test.go
+++ b/schema/table_test.go
@@ -186,6 +186,7 @@ func TestTable(t *testing.T) {
 
 		table := tables.Get(reflect.TypeFor[*Embeded]())
 		require.Contains(t, table.StructMap, "profile")
+		require.Equal(t, table.StructMap["profile"].Table.Name, "profiles")
 	})
 
 	t.Run("embed scanonly", func(t *testing.T) {

